runtime/altair/src/migrations.rs

```diff
@@ -890,3 +892,27 @@ mod pool_system {
 		}
 	}
 }
+
+mod xcm_v2_to_v3 {
+	use super::*;
+	use crate::{PolkadotXcm, RuntimeOrigin, VERSION};
+
+	pub struct SetSafeXcmVersion;
+
+	impl OnRuntimeUpgrade for SetSafeXcmVersion {
+		fn on_runtime_upgrade() -> Weight {
+			if VERSION.spec_version > 1030 {
+				return Weight::zero();
+			}
+
+			// Unfortunately, SafeXcmVersion storage is not leaked to runtime
+			PolkadotXcm::force_default_xcm_version(
+				RuntimeOrigin::root(),
+				Some(cfg_primitives::SAFE_XCM_VERSION),
+			)
+			.unwrap_or_else(|_| log::error!("Failed to set safe XCM version on runtime upgrade, requires manual call via governance"));
+
+			RocksDbWeight::get().reads_writes(1, 1)
+		}
+	}
```

Comment:
No `before` and `after`?

Reply:
See the comment before setting:
> Unfortunately, SafeXcmVersion storage is not leaked to runtime

So we can't check this storage in `pre` and `post` upgrade unfortunately. Since we have no `PolkadotXcm` storage at the moment, we can't batch the runtime upgrade with the execution of `force_default_xcm_version` either unfortunately. 

Either we live with the fact it cannot be checked with `try-runtime` or need to create a post-upgrade referendum to set it. Given that we only do a single write here, I think it is sane to keep the migration. I have tested it locally against `centrifuge-local` @ `1019`.

Reply:
Sorry, yeah, thats true. Lets keep it as is